File: Src/java/engine/src/test/java/org/opencds/cqf/cql/engine/execution/Issue33.java
package org.opencds.cqf.cql.engine.execution;

import static org.junit.jupiter.api.Assertions.assertTrue;

import java.math.BigDecimal;
import org.junit.jupiter.api.Test;
import org.opencds.cqf.cql.engine.elm.executing.EquivalentEvaluator;
import org.opencds.cqf.cql.engine.runtime.DateTime;
import org.opencds.cqf.cql.engine.runtime.Interval;

class Issue33 extends CqlTestBase {

    @Test
    void interval() {
        final BigDecimal bigDecimalZoneOffset = getBigDecimalZoneOffset();

        var results = engine.evaluate(toElmIdentifier("Issue33"));
        Object value = results.forExpression("Issue33").value();
        assertTrue(EquivalentEvaluator.equivalent(
                ((Interval) value).getStart(), new DateTime(bigDecimalZoneOffset, 2017, 12, 20, 11, 0, 0)));
        assertTrue(EquivalentEvaluator.equivalent(
                ((Interval) value).getEnd(), new DateTime(bigDecimalZoneOffset, 2017, 12, 20, 23, 59, 59, 999)));
    }
}

File: Src/java/engine/src/jvmTest/kotlin/org/opencds/cqf/cql/engine/execution/Issue33.kt
package org.opencds.cqf.cql.engine.execution

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.opencds.cqf.cql.engine.elm.executing.EquivalentEvaluator
import org.opencds.cqf.cql.engine.runtime.DateTime
import org.opencds.cqf.cql.engine.runtime.Interval

internal class Issue33 : CqlTestBase() {
    @Test
    fun interval() {
        val bigDecimalZoneOffset = bigDecimalZoneOffset

        val results = engine.evaluate { library("Issue33") }.onlyResultOrThrow
        val value = results["Issue33"]!!.value
        Assertions.assertTrue(
            EquivalentEvaluator.equivalent(
                (value as Interval).start,
                DateTime(bigDecimalZoneOffset, 2017, 12, 20, 11, 0, 0),
            ) == true
        )
        Assertions.assertTrue(
            EquivalentEvaluator.equivalent(
                value.end,
                DateTime(bigDecimalZoneOffset, 2017, 12, 20, 23, 59, 59, 999),
            ) == true
        )
    }
}
